### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by dimbleby**
A series of commits tightening up the type-checking in various places

### CODE REVIEW

File: src/poetry/console/commands/cache/clear.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def handle(self) -> int:
                     f"Add the --all option if you want to clear all {parts[0]} caches"
                 )
 
-            if not os.path.exists(str(cache_dir)):
+            if not os.path.exists(cache_dir):
```
Comment (Secrus) [inline src/poetry/console/commands/cache/clear.py:49]: Why not use `pathlib.Path.exists()`? Seems cleaner.

File: src/poetry/utils/authenticator.py [hunk 2]
```diff
@@ -202,18 +194,17 @@ def request(
...
+        proxies: dict[str, str] = kwargs.get("proxies", {})
+        stream: bool | None = kwargs.get("stream")
 
         certs = self.get_certs_for_url(url)
-        verify = kwargs.get("verify") or certs.cert or certs.verify
-        cert = kwargs.get("cert") or certs.client_cert
+        verify: bool | str | Path = kwargs.get("verify") or certs.cert or certs.verify
+        verify = str(verify) if isinstance(verify, Path) else verify
 
+        cert: str | Path | None = kwargs.get("cert") or certs.client_cert
```
Comment (Secrus) [inline src/poetry/utils/authenticator.py:204]:
```suggestion
        cert: str | Path | None = kwargs.get("cert", certs.client_cert)
```
Isn't this the same?

**Review by Secrus — CHANGES_REQUESTED** [review]

File: src/poetry/utils/authenticator.py (hunk 2 above)
Comment (dimbleby) [reply src/poetry/utils/authenticator.py:204]: not _quite_ the same, if `cert` is somehow an empty string...

Comment (Secrus) [reply src/poetry/utils/authenticator.py:204]: Oh, right, didn't think about this case.

**Review by Secrus — APPROVED** [review]
LGTM, but let's get a second opinion

**Comment by dimbleby** [comment]
are we waiting for a second opinion from anyone in particular?  do they know that?

**Review by branchvincent — APPROVED** [review]

**Pull request merged**